### PR TITLE
Softfork2 infrastructure

### PR DESCRIPTION
**Pull request by arvidn**
introduce the new constant for the new upcoming soft-fork (for 1.8.0). Also enable parsing the `ASSERT_BEFORE_*` conditions after that height.

This flag also enables the other new conditions like `ASSERT_CONCURRENT_SPEND` and `ASSERT_CONCURRENT_PUZZLE`.

Note that these are already enabled in `MEMPOOL_MODE` unconditionally.

### CODE REVIEW

File: tests/core/test_cost_calculation.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ async def test_mempool_mode(self, softfork_height, bt):
         program = SerializedProgram.from_bytes(
             binutils.assemble(
                 f"(q ((0x3d2331635a58c0d49912bc1427d7db51afe3f20a7b4bcaffa17ee250dcbcbfaa {disassembly} 300"
-                f"  (() (q . ((65 '00000000000000000000000000000000' 0x0cbba106e000))) ()))))"
+                f"  (() (q . ((48 '00000000000000000000000000000000' 0x0cbba106e000))) ()))))"
```
Comment (aqk) [inline tests/core/test_cost_calculation.py:144]: What does this line do? ('A' -> '0')

Comment (richardkiss) [reply tests/core/test_cost_calculation.py:144]: I also cannot figure this out.

**Comment by richardkiss** [comment]
This seems simple enough.

File: tests/core/test_cost_calculation.py (hunk 1 above)
Comment (arvidn) [reply tests/core/test_cost_calculation.py:144]: I should have highlighted this change. the hint is in the commit message for this one change. This test is checking that unknown condition codes are ignored in normal consensus mode but rejected in mempool mode. The issue is that with this softfork, 65 is now a *known* condition code. This is just changing it to still be something unknown

**Review by richardkiss — APPROVED** [review]

**Review by trepca — APPROVED** [review]

**Pull request merged**